kernel/context.ml

```diff
@@ -179,6 +179,10 @@ struct
       | LocalAssum (na, ty) -> na, None, ty
       | LocalDef (na, v, ty) -> na, Some v, ty
 
+    let of_tuple = function
```

Comment:
I'd rather not have this function added. The reason we have the dual one is due to historical reasons, in order to easily port code using the previous API. Rather, the only user introduced in this PR should match on the argument.

Reply:
I agree. So, just to be sure, do you mean directly passing a decl to add_name, rather than deconstructing/reconstructing?

PS: There is already an of_tuple for named decl!

Reply:
@herbelin yes, the caller code seems easily amenable to using constructors directly.